doc/command-line.md

```diff
@@ -758,3 +782,113 @@ For example,
 sourmash signature overlap file1.sig file2.sig
 ```
 will display the detailed comparison of `file1.sig` and `file2.sig`.
+
+## Advanced command-line usage
+
+### Loading signatures and databases
+
+sourmash uses several different command-line styles.
+
+Briefly,
+
+* `search` and `gather` both take a single query signature and search
+  multiple signatures or databases. In this case, there has to be a
+  single identifiable query for sourmash to use, and if you're using a
+  database or list of signatures as the source of a query, you'll
+  need to provide a selector (ksize with `-k`, moltype with `--dna` etc,
+  or md5sum with `--query-md5`) that picks out a single signature.
+
+* `compare` takes multiple signatures and can load them from files,
+  directories, and indexed databases (SBT or LCA).  It can also take
+  a list of file paths in a text file, using `--from-file` (see below).
+  
+* the `lca classify` and `lca summarize` commands take multiple
+  signatures with `--query`, and multiple LCA databases, with
+  `--db`. `sourmash multigather` also uses this style.  This allows these
+  commands to specify multiple queries **and** multiple databases without
+  (too much) confusion.  These commands will take files containing
+  signature files using `--query-from-file` (see below).
+  
+* `index` and `lca index` take a few fixed parameters (database name,
+  taxonomy spreadsheet) and then an arbitrary number of other files
+  that contain signatures, including files, directories, and indexed
+  databases. These commands will also take `--from-file` (see below).
+
+#### Storing (and searching) signatures
+  
+Backing up a little, there are many ways to store and search
+signatures.
+
+The simplest is one signature in a single JSON file. You can also put
+many signatures in a single JSON file, either by building them that
+way with `sourmash compute` or by using `sourmash sig cat` or other
+commands. Searching or comparing these files involves loading them
+sequentially and iterating across all of the signatures - which can be
+slow, especially for many (100s or 1000s) of signatures.
+
+Indexed databases can make searching signatures a lot faster. SBT
+databases are low memory and disk-intensive databases that allow for
+fast searches using a tree structure, while LCA databases are higher
+memory and (after a potentially significant load time) are quite fast.
+
+(LCA databases also permit taxonomic searches using `sourmash lca` functions.)
+
+The main point is that since all of these databases contain signatures,
+as of sourmash 3.4, any command that takes more than one signature will
+also automatically load all of the signatures in the database.
+
+Note that, for now, both SBT and LCA database can only contain one
+"type" of signature (one ksize, one moltype, etc.) If the database
+signature type is incompatible with the other signatures, sourmash
+will complain.  will be raised.  In contrast, signature files can
```

Comment:
```suggestion
will complain. In contrast, signature files can
```

Unless you wanted to mention the specific error?

Reply:
The error message should be self-explanatory, but I'll check.

Reply:
whoops, nice catch, found a bug 😂